src/OneMightyRoar/PHP_ActiveRecord_Components/Cache/Adapter/DoctrineCacheAdapter.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * PHP-ActiveRecord-Components
+ *
+ * Useful common components for a php-activerecord based project
+ *
+ * @copyright   2013 One Mighty Roar
+ * @link        http://onemightyroar.com
+ */
+
+namespace OneMightyRoar\PHP_ActiveRecord_Components\Cache\Adapter;
+
+use Doctrine\Common\Cache\CacheProvider;
+use OneMightyRoar\PHP_ActiveRecord_Components\Cache\CacheAdapterInterface;
+
+/**
+ * DoctrineCacheAdapter
+ *
+ * An adapter to translate a Doctrine {@see CacheProvider} into a PHP-ActiveRecord
+ * compatible cache adapter, through the {@see CacheAdapterInterface}
+ *
+ * @package OneMightyRoar\PHP_ActiveRecord_Components\Cache\Adapter
+ */
+class DoctrineCacheAdapter implements CacheAdapterInterface
+{
+
+    /**
+     * Constants
+     */
+
+    /**
+     * The default value for the option of whether or not to flush the namespace
+     * instead of the entire cache for flush operations
+     *
+     * @type bool
+     */
+    const FLUSH_NAMESPACE_DEFAULT = false;
+
+
+    /**
+     * Properties
+     */
+
+    /**
+     * The Doctrine cache provider to be adapted
+     *
+     * @type CacheProvider
+     */
+    private $adapted_provider;
+
+    /**
+     * Whether or not to flush the namespace instead of the entire cache when
+     * committing a cache flush
+     *
+     * @type bool
+     */
+    private $flush_namespace = false;
+
+
+    /**
+     * Methods
+     */
+
+    /**
+     * Constructor
+     *
+     * @param CacheProvider $cache_provider The Doctrine cache provider to be adapted
+     */
+    public function __construct(CacheProvider $cache_provider, $flush_namespace = self::FLUSH_NAMESPACE_DEFAULT)
+    {
+        $this->adapted_provider = $cache_provider;
+        $this->flush_namespace = (bool) $flush_namespace;
+    }
+
+    /**
+     * Get the value of flush_namespace
+     *
+     * @access public
+     * @return bool
+     */
+    public function getFlushNamespace()
+    {
+        return $this->flush_namespace;
+    }
+
+    /**
+     * Set the value of flush_namespace
+     *
+     * @param bool $flush_namespace
+     * @access public
+     * @return $this
+     */
+    public function setFlushNamespace($flush_namespace)
```

Comment:
Yea, I don't feel strongly about it either way. I _could_ make the property `protected` (instead of `private`) and remove the setter, to allow for easier change by an extension. Not sure.